### PR TITLE
[drape] Remove redundant Freetype calls

### PR DESCRIPTION
**Pull request by biodranik**
Rendered fonts look exactly the same after this change.

This is a pre-requisite for https://github.com/organicmaps/organicmaps/issues/4281

### CODE REVIEW

File: drape/glyph_manager.cpp [hunk 1]
```diff
@@ -175,14 +170,18 @@ class Font
...
-    Glyph result;
-    result.m_image = {bitmap.width, bitmap.rows, data};
     // Glyph image has SDF borders that should be taken into an account.
-    result.m_metrics = {static_cast<float>(glyph->advance.x >> 16), static_cast<float>(glyph->advance.y >> 16),
-                        static_cast<float>(bbox.xMin + kSdfBorder), static_cast<float>(bbox.yMin + kSdfBorder), true};
-    result.m_code = unicodePoint;
-    FT_Done_Glyph(glyph);
-
+    float const xAdvance = glyph->advance.x >> 6;
+    float const yAdvance = glyph->advance.y >> 6;
```
Comment (vng) [inline drape/glyph_manager.cpp:175]: Before 16, now 6. Is there any logic in these numbers or some docs? Is it possible to check/test at least where these numbers are not zeros?

File: drape/glyph_manager.cpp [hunk 2]
```diff
@@ -175,14 +170,18 @@ class Font
...
+    float const xOffset = glyph->metrics.horiBearingX >> 6;
+    // yOffset uses bottom left coordinate in calculations.
+    float const yOffset = (glyph->metrics.horiBearingY - glyph->metrics.height) >> 6;
```
Comment (vng) [inline drape/glyph_manager.cpp:178]: What about kSdfBorder?

File: drape/glyph_manager.cpp (hunk 1 above)
Comment (biodranik) [reply drape/glyph_manager.cpp:175]: Freetype operates with the 26.6 fixed-point format, where each point represents 1/64th of a pixel, see https://freetype.org/freetype2/docs/glyphs/glyphs-3.html

Not sure why the previous code used 16-bit shifts.

Comment (biodranik) [reply drape/glyph_manager.cpp:175]: See also https://freetype.org/freetype2/docs/reference/ft2-glyph_management.html#ft_glyph_get_cbox

Comment (vng) [reply drape/glyph_manager.cpp:175]: Well, I can agree here and merge, but honestly, I'd like to see what happens with glyphs with advance != 0 before and after the fix :)

Comment (biodranik) [reply drape/glyph_manager.cpp:175]: Every glyph always has xAdvance set for horizontal layouts. Any wrong xAdvance is immediately seen on the screen as an incorrect distance between glyphs.

Comment (biodranik) [reply drape/glyph_manager.cpp:175]: Freetype multiplies it for some reason in the retrieved glyph, but does not multiply it in the face's internal glyph:

выява

Comment (biodranik) [reply drape/glyph_manager.cpp:175]: I think it's 16.16 points coordinates, vs 26.6 points coordinates. Here:
https://freetype.org/freetype2/docs/reference/ft2-glyph_management.html#ft_get_glyph
выява

**Review by vng — APPROVED** [review]

**Pull request merged**